### ISSUE TITLE
App crashes when switching to "Feed" tab

### ISSUE THREAD

**Issue by webdevsuperfast** [ram-on/SkyTube#594]
Hi,

Just updated to SkyTube Extra v2.966 and noticed that when navigating/switching to `Feed` tab, the app crashes.

**Device:** Asus Zenfone Max Pro M1
**OS Version:** Android 9

Thanks!

**Comment by atomjack** [comment]
I'm getting this, too:

```     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'long java.lang.Long.longValue()' on a null object reference
        at free.rm.skytube.businessobjects.db.Tasks.GetSubscriptionsVideosFromDb.getNextVideos(GetSubscriptionsVideosFromDb.java:53)
        at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:88)
        at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:40)
```

It looks to be caused by [this recent change](https://github.com/ram-on/SkyTube/commit/0612d5bc500ca33ff72d8c7bd65f64a04f5b75db#diff-a1f5e663b69ca51646cda6160f608a22). @gzsombor Do you know what this is about?

I temporarily undid that change on my end and it loads fine now.

**Comment by gzsombor** [comment]
Is it a NullPointerException coming from `free.rm.skytube.businessobjects.db.Tasks.GetSubscriptionsVideosFromDb.getNextVideos(GetSubscriptionsVideosFromDb.java:53)` ?

I haven't seen this error yet, but I guess, this is related the content of your db, maybe some field were missing, or incorrectly initialized.

**Comment by atomjack** [comment]
Yeah, it's from this line:
`				lastVideoPublishTimestamp = last.getRetrievalTimestamp();//last.getPublishDate().getValue();`

It's because YouTubeVideo.retrievalTimestamp is null. This is a new field, and I'm assuming the videos stored in the database do not have this field defined. The videos in the database were defined before this field was added to the code.

**Comment by gzsombor** [comment]
Strange, because I've added a code to backfill from the getPublishDate, [here](https://github.com/ram-on/SkyTube/blob/0612d5bc500ca33ff72d8c7bd65f64a04f5b75db/app/src/main/java/free/rm/skytube/businessobjects/db/SubscriptionsDb.java#L118)
Either it's not executed - I highly doubt - or there were no getPublishDate stored in the db...

**Comment by webdevsuperfast** [comment]
@gzsombor it seems working if you subscribe to a new channel.

**Comment by Tvax** [comment]
I just updated to this version from FDroid. And the problem is that my launching tab is set to the "Feed" one. So my app crashes at startup.

**Comment by Tvax** [comment]
Made a fix #595

**Comment by gzsombor** [comment]
I've attached a zip file, which contains a patched SkyTube Extra and SkyTube Oss, could you please try it? (I guess, to install the SkyTube OSS version to overwrite your version from F-Droid wont work, but maybe you can try it...)

[fix-594.zip](https://github.com/ram-on/SkyTube/files/3799871/fix-594.zip)

**Comment by ZenMasta** [comment]
I probably should have checked here before updating. And I even waited a few days 😅

**Comment by Tvax** [comment]
Yeah this is not signed with the same signature as the FDroid one so I can't really test it as I didn't export my subscriptions database beforehand.
Maybe you should try publishing this hotfix directly through FDroid. Worst case scenario it doesn't work, considering the current state of the app it should not bother anyone.

**Comment by gzsombor** [comment]
I can try it, but publishing to FDroid is slow - they hadn't merged 2.965, when 2.966 was released, so that's why no 2.965 on FDroid.

**Comment by gzsombor** [comment]
For me, this error disappears with the latest 2.967 build, I'm going to open a ticket for F-Droid as well.
